### ISSUE TITLE
make.py should not use `os.system("sphinx-build ...")`

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#8208]
I know the plan is to switch to a doc makefile, but the issue would be the same.

Calling `os.system("sphinx-build ...")` (or directly `sphinx-build` in the Makefile) will pick up a sphinx install outside of the current environment, which may or may not have an unrelated (e.g., older) version of matplotlib installed.  Instead, the command should be something like `python -msphinx build ...` if sphinx offers that option, or we should use a wrapper script otherwise that makes sure that sphinx is used from the current environment only.

**Comment by matthew-brett** [comment]
Looking at the contents of `sphinx-build`, you could probably get away with something like:

```
import sphinx
sphinx.build_main(['sphinx-build'] + input_args)
```

**Comment by QuLogic** [comment]
I don't understand your concern; why would it pick up `sphinx-build` outside the environment? If the environment didn't install the executable and modify `PATH`, then something is wrong with the environment.

There _is_ a problem on Python 3 though _if_ using system packages, because they generally install both `sphinx-build` and `sphinx-build-3`. This is handled by the Makefile using a modifiable variable `SPHINXBUILD` which can be set when calling make: `make SPHINXBUILD=sphinx-build-3`.

**Comment by matthew-brett** [comment]
It's quite common in my experience for me to have `sphinx-build` installed on the system, or in my user installs, and for that to get picked up when I am working inside a virtualenv, on which I have not installed sphinx.   That can be pretty confusing.

**Comment by QuLogic** [comment]
Oh yes, not having it installed at all, good point. We can fix this now, but I think there's not really much of a way to fix this when switching to the Makefile, though.

**Comment by anntzer** [comment]
Actually, as of sphinx 1.5.3, you can just do
```
python -msphinx -bhtml $srcdir $destdir
```
Perhaps we can just check since when this form is supported and update the requirement.  I will open an issue upstream to fix their Makefile/make.bat as well.

**Comment by anntzer** [comment]
xref https://github.com/sphinx-doc/sphinx/pull/3523

**Comment by anntzer** [comment]
Closed by #8269.
